### ISSUE TITLE
Error: The terraform-provider-prismacloud_v1.4.1 plugin crashed!

### ISSUE THREAD

**Issue by judiethel** [PaloAltoNetworks/terraform-provider-prismacloud#235]
## Describe the bug

While attempting to create a Prisma alert rule using Terraform, the plan stage completed successfully, but the apply stage resulted in an interface conversion error. The error message indicated that there was an attempt to convert a nil value to a map[string]interface{}, and provided a stack trace that pointed to the parseAlertRule function in the resource_alert_rule.go file of the terraform-provider-prismacloud_v1.4.1 plugin.

When I remove this attributes the resource will be created. So there must be a bug within this part.
```
  resource_list {
      compute_access_group_ids = []
    }
```

From what I understand:

* If resource_list is defined, not null and not empty do (resource_list is considered as a list, as you can have multiple blocks)
* Cast resource list as a list
* create a new list of same size as resource_list
* Get first element of the original list and cast is as a map
* Add this element to the new list

But it's casted as a map of string not a map of list
```
	if rle := tgt["resource_list"]; rle != nil && len(rle.([]interface{})) > 0 {
		rl := rle.([]interface{})
		resourceList = make([]rule.ResourceList, 0, len(rl))
		rmap := rl[0].(map[string]interface{})
		resourceList = append(resourceList, rule.ResourceList{
			IncludedResourceLists: ListToStringSlice(rmap["compute_access_group_ids"].([]interface{})),
		})
	}
```

## Expected behavior

Normally I except that a new Prisma Alert rule will be added to the Cloud. 

## Screenshots

![image](https://github.com/PaloAltoNetworks/terraform-provider-prismacloud/assets/102523435/862a5dad-b8a2-400d-aa19-dd4d142ee566)

## Error message

```
Stack trace from the terraform-provider-prismacloud_v1.4.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 129 [running]:
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud.parseAlertRule(_, {_, _})
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud/resource_alert_rule.go:396 +0x12b7
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud.createAlertRule({0xd02928, 0xc000294de0}, 0xc0007b7790, {0xb95700, 0xc0001c05a0})
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud/resource_alert_rule.go:583 +0x8d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0004a0780, {0xd028b8, 0xc000126740}, 0x2, {0xb95700, 0xc0001c05a0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/resource.go:285 +0x1a3
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0004a0780, {0xd028b8, 0xc000126740}, 0xc0001981c0, 0xc00075dec0, {0xb95700, 0xc0001c05a0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/resource.go:396 +0x86f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001244f8, {0xd028b8, 0xc000126740}, 0xc0001ec0a0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/grpc_provider.go:955 +0x9aa
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00011f960, {0xd02960, 0xc0003a61b0}, 0x7f34a720b280)
github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:332 +0x6c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xbc73c0, 0xc00011f960}, {0xd02960, 0xc0003a61b0}, 0xc000198000, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000394540, {0xd0f5f8, 0xc000507380}, 0xc0003f0000, 0xc000534a50, 0x11a3a80, 0x0)
google.golang.org/grpc@v1.48.0/server.go:1295 +0xb03
google.golang.org/grpc.(*Server).handleStream(0xc000394540, {0xd0f5f8, 0xc000507380}, 0xc0003f0000, 0x0)
google.golang.org/grpc@v1.48.0/server.go:1636 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.48.0/server.go:930 +0x294

Error: The terraform-provider-prismacloud_v1.4.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
## Your Environment
- Terraform version: 1.5.0
- Provider Version: 1.4.1

**Comment by landro** [comment]
We're seeing this exact issue too! Same environment as @judiethel

**Comment by ftbrecordspan** [comment]
This is fixed under v1.4.2.
